src/server/context.ts

```diff
@@ -400,6 +400,16 @@ export class ServerContext {
           if (route.component === undefined) {
             throw new Error("This page does not have a component to render.");
           }
+
+          if (
+            typeof route.component === "function" &&
+            route.component.constructor.name === "AsyncFunction"
+          ) {
+            throw new Error(
+              "Async components are not supported. Fetch data inside of the loader, as described in the docs: https://fresh.deno.dev/docs/getting-started/fetching-data",
```

Comment:
https://github.com/denoland/fresh/pull/413#discussion_r913696364

```suggestion
              "Async components are not supported. Fetch data inside of a route handler, as described in the docs: https://fresh.deno.dev/docs/getting-started/fetching-data",
```